pkg/kubeapiserver/default_storage_factory_builder.go

```diff
@@ -78,6 +80,14 @@ func NewStorageFactoryConfig() *StorageFactoryConfig {
 		storage.Resource("volumeattributesclasses").WithVersion("v1alpha1"),
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.EmulationVersion) {
+		// If emulation versioning is enabled, then the storage version depends
+		// on the user's configured emulation version and the mincompatibilityversion
+		// If mincompatibilityVersion is N-1, then the results should be the
+		// same as the old hardcoded list.
+		resources = nil
+	}
+
```

Comment:
@jpbetz is this logic sound? I think it is but it is seeming like we wont include lifecycle info for pre-existing resources? Im wondering how this override list should reconcile with `--emulated-version`

Reply:
It's great that we're going to be able to remove the manually managed list in the future. That's a nice net win.

Reply:
Yeah though I think it cannot go in this alpha since we are lacking lifecycle information for these resources

Reply:
Yep, I agree.